### ISSUE TITLE
The event payload for `battery_notes_battery_threshold` does not set `previous_battery_level` to a float as expected.

### ISSUE THREAD

**Issue by uvjim** [andrew-codechimp/HA-Battery-Notes#1236]
## System Information

version | core-2024.2.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.1
os_name | Linux
os_version | 6.1.73-haos-raspi
arch | aarch64
timezone | Europe/London
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1409
Downloaded Repositories | 24
HACS Data | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.0
-- | --
update_channel | stable
supervisor_version | supervisor-2024.02.1
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 116.5 GB
disk_used | 11.4 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.8.0), Samba share (12.3.0), Zigbee2MQTT (1.36.0-1), Nextcloud Backup (0.17.7), Advanced SSH & Web Terminal (17.1.1), Mosquitto broker (6.4.0), ESPHome (2024.2.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 16
views | 12
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 26 February 2024 at 14:03
-- | --
current_recorder_run | 5 March 2024 at 14:46
estimated_db_size | 687.79 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

### Checklist

- [X] I have added `battery_notes:` to my configuration.yaml and restarted.
- [X] I have read the [FAQ's](https://andrew-codechimp.github.io/HA-Battery-Notes/faq).
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

The event payload for `battery_notes_battery_threshold` does not set `previous_battery_level` to a float as expected.

```yaml
this:
  entity_id: automation.notification_battery_wip
  state: 'on'
  attributes:
    id: '1709646966958'
    last_triggered: '2024-03-05T16:43:16.020291+00:00'
    mode: single
    current: 0
    friendly_name: 'Notification: Battery (WIP)'
  last_changed: '2024-03-05T16:23:14.862840+00:00'
  last_updated: '2024-03-05T16:43:16.072640+00:00'
  context:
    id: 01HR7QYY5KM8NGNP5026JRTP45
    parent_id: 01HR7QYY5FGK3VZ3QFJVFEABZ1
    user_id: null
trigger:
  id: Battery Above Threshold
  idx: '1'
  alias: Battery Above Threshold
  platform: event
  event:
    event_type: battery_notes_battery_threshold
    data:
      device_id: db3c9b28cd6098784634cbaf4d1ea39f
      device_name: James' Phone
      battery_low: false
      battery_type_and_quantity: Not replaceable
      battery_type: Not replaceable
      battery_quantity: 1
      battery_level: 31
      previous_battery_level: '29'
      reminder: false
    origin: LOCAL
    time_fired: '2024-03-05T16:44:39.876764+00:00'
    context:
      id: 01HR7R1G24CA2N2MXB5SY0RNTT
      parent_id: null
      user_id: null
  description: event 'battery_notes_battery_threshold'
```

### Reproduction steps

Look at the payload of the `battery_notes_battery_threshold` event once it is fired.

**Comment by andrew-codechimp** [comment]
Good spot, I'll get that into the next release.

**Comment by andrew-codechimp** [comment]
This is now fixed in beta 2.2.1

**Comment by uvjim** [comment]
Confirmed fixed.
